tools/container/nvidia-toolkit/run.go

```diff
@@ -182,9 +201,16 @@ func verifyFlags(o *options) error {
 	return nil
 }
 
-func initialize() error {
+func initialize(pidFile string) error {
```

Comment:
Is it possible to run this as an `init()` function instead? In this case, it would be guaranteed to be run only once and we can access `o.pidFile` globally instead of passing it as an argument

Reply:
We can't run it as an init function because `o.pidFile` is only set when the CLI application is run and the relevant flags are parsed. It is my understanding that the `init` function is invoked before this.

Assuming that we prescribe the path, then we could do it but we would have to also include logic to close the file and release the exclusive lock.